spec/p2p/reactor-survey.md

```diff
@@ -0,0 +1,247 @@
+
+As 
+- full nodes add transactions to the mempool following client requests, and
+- validators read transactions from the mempool to compose proposed blocks,
+
+the mentioned end-to-end requirements on transactions translate into requirements 
+on the mempool, e.g., every transaction that is submitted to a full node should 
+eventually be present at the mempool of a validator/proposer. 
+
+For the purpose of this document (specifying what the mempool needs from p2p), 
+we do not strive to precisely state the end-to-end requirements. Rather, we 
+will show that the arguably strongest requirement on the mempool do translate roughly
+into requirements on the p2p layer that are similar to those of the
+consensus reactor.
+
+- end-to-end requirement: "every transaction submitted should be eventually put into a block"
+> leads to
+- requirement for consensus: for every submitted transaction from some height on, 
+the transaction is proposed by every correct proposer in every height until it is committed
+into a block (assuming infinitely often correct proposers get their proposal through)  
+> leads to
+- current solution to address this requirement: 
+    - the mempool of correct full nodes is organized as a list. 
+    This list can be understood as a shared data structure with relaxed semantics, that is,
+    not all nodes need to agree on the order of the elements, but we have a property that 
+    approximately says "if at time *t* a transaction tx1 is in the list of **all** nodes, and a 
+    client submits transaction tx2 after time t, then it is never the case the tx2 is before
+    tx1 in a list of a correct validator.
+
+    > We talk about "correct full nodes" here as it describes "the current solution", the
+    way this solution contributed to solving the problem is that if we have a property on
+    every correct full node, this implies that we have this property on every (potential) 
+    validator, and consequently, every correct proposer.
+
+    - to prepare a proposal,  a validator requests from the mempool a list of
+pending transactions that respects certain limits, in terms of the number of
+transactions returned, their total size in bytes, and their required gas.
+The mempool then returns the longest **prefix** of its local list of pending
+transactions that respects the limits established by the consensus protocol. 
+
+  > As a result of these two properties, once a transaction is known to all validators (present and future), it 
+cannot be overtaken by infinitely many transaction that arrived in the system later (there
+is still some possible overtaking due to faulty proposers); cf. no starvation.
+
+
+We arrive at the requirement on the mempool: 
+
+#### [CM-REQ-MEMP-GOSSIP.0]
+
+1. every transaction submitted to a correct full node must eventually be included in the mempool of a correct validator
```

Comment:
We should figure out whether we want the same guarantees for transactions submitted to any fullnode, of e.g. for MEV, you submit your transaction to a special full node (or a set of full nodes) and you get better guarantees. Best effort transactions vs. transactions with guarantees.
Might be a question for the mempool spec.

Reply:
Daniel: Define stable nodes vs. transient nodes is important. We can give guarantees wrt stable nodes, while we allow transient nodes.